components/places/ffi/src/lib.rs

```diff
@@ -341,6 +343,115 @@ pub unsafe extern "C" fn sync15_history_sync(
     })
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn bookmarks_get_tree(
+    handle: u64,
+    optional_root_id: *const c_char,
+    error: &mut ExternError,
+) -> ByteBuffer {
+    log::debug!("bookmarks_get_tree");
+    CONNECTIONS.call_with_result(error, handle, |conn| -> places::Result<_> {
```

Comment:
As above, I'm inclined to think that not suppyling a root should be an error rather than defaulting to a potentially expensive operation.